packages/spatial/src/transform/components/EntityTree.tsx

```diff
@@ -494,17 +494,23 @@ export function useChildWithComponents(rootEntity: Entity, components: Component
   return result.value
 }
 
-export function useChildrenWithComponents(rootEntity: Entity, components: ComponentType<any>[]): Entity[] {
+export function useChildrenWithComponents(
+  rootEntity: Entity,
+  components: ComponentType<any>[],
+  exclude: ComponentType<any>[] = []
+): Entity[] {
   const children = useHookstate([] as Entity[])
   const componentsString = components.map((component) => component.name).join()
+  const excludeString = exclude.map((component) => component.name).join()
   useLayoutEffect(() => {
     let unmounted = false
     const ChildSubReactor = (props: { entity: Entity }) => {
       const tree = useOptionalComponent(props.entity, EntityTreeComponent)
       const matchesQuery = _useHasAllComponents(props.entity, components)
+      const matchesExludeQuery = _useHasAllComponents(props.entity, exclude) && exclude.length
```

Comment:
i think this would have to be a `_useHasAnyComponents` which would OR them all, not AND, otherwise [Component1, Component2] would only exclude if both were there, not if just one or none was on the entity